code/game/machinery/_machinery.dm

```diff
@@ -1029,3 +1029,33 @@
 	if(isliving(user))
 		last_used_time = world.time
 		last_user_mobtype = user.type
+
+/**
+ * Causes the machinery to shake while in use.
+ *
+ * Uses an animation to cause the machinery to shake for a designated amount of time.
+ */
+/obj/machinery/proc/start_shaking()
```

Comment:
we have an atom proc called Shake

Reply:
The `/atom/proc/Shake()` doesn't quite do what's seems visually correct as it uses duration in terms of repetitions instead of time elapsed, as the `shake_for` proc does.

I can try and write a way to translate the duration into the loop for the atom proc.

Reply:
The atom ``Shake()`` proc seems fine and does have a timer arg, you can feel free to improve it if you don't like how it handles it.

Reply:
https://github.com/tgstation/tgstation/pull/68885/commits/b52efb5fe2c8feac1b859f2df91707d7367054b7 @JohnFulpWillard